src/routes/jira/security/workspaces/jira-security-workspaces-post.test.ts

```diff
@@ -0,0 +1,324 @@
+import { getLogger } from "config/logger";
+import express, { Application } from "express";
+import { getFrontendApp } from "~/src/app";
+import supertest from "supertest";
+import { Installation } from "models/installation";
+import { createQueryStringHash, encodeSymmetric } from "atlassian-jwt";
+import { Errors } from "config/errors";
+import { Subscription } from "models/subscription";
+import { RepoSyncState } from "models/reposyncstate";
+import { DEFAULT_AVATAR } from "routes/jira/security/workspaces/jira-security-workspaces-post";
+import { envVars } from "~/src/config/env";
+
+const createMultipleSubscriptionsAndRepos = async () => {
+	const sub1 = await Subscription.create({
+		id: 1234,
+		gitHubInstallationId: "1234",
+		jiraHost:envVars.APP_URL,
+		jiraClientKey: "client-key",
+		avatarUrl: "http://myavatarurl"
+	});
+
+	const sub2 = await Subscription.create({
+		id: 2345,
+		gitHubInstallationId: "2345",
+		jiraHost:envVars.APP_URL,
+		jiraClientKey: "client-key",
+		avatarUrl: "http://anotheravatarurl"
+	});
+
+	const sub3 = await Subscription.create({
+		id: 3456,
+		gitHubInstallationId: "3456",
+		jiraHost:envVars.APP_URL,
+		jiraClientKey: "client-key"
+	});
+	const sub4 = await Subscription.create({
+		id: 678,
+		gitHubInstallationId: "678",
+		jiraHost:envVars.APP_URL,
+		jiraClientKey: "client-key"
+	});
+
+	const repo1 = await RepoSyncState.create({
+		subscriptionId: sub1.id,
+		repoId: 1,
+		repoName: "repo",
+		repoOwner: "owner",
+		repoFullName: "owner/repo",
+		repoUrl: "https://github.com/owner/repo"
+	});
+
+	const repo2 = await RepoSyncState.create({
+		subscriptionId: sub2.id,
+		repoId: 2,
+		repoName: "my-repo",
+		repoOwner: "atlassian",
+		repoFullName: "atlassian/my-repo",
+		repoUrl: "https://github.internal.atlassian.com/atlassian/my-repo"
+	});
+
+	const repo3 = await RepoSyncState.create({
+		subscriptionId: sub3.id,
+		repoId: 3,
+		repoName: "repo3",
+		repoOwner: "owner3",
+		repoFullName: "owner3/repo3",
+		repoUrl: "https://github.com/owner3/repo3"
+	});
+
+	return { sub1, sub2, sub3, sub4, repo1, repo2, repo3 };
+};
+
+describe("Workspaces Post", () => {
+	let app: Application;
+	let installation: Installation;
+	let createdDbEntries;
+
+
+	beforeEach(async () => {
+		createdDbEntries = await createMultipleSubscriptionsAndRepos();
+		installation = await Installation.install({
+			host: jiraHost,
+			sharedSecret: "shared-secret",
+			clientKey: "jira-client-key"
+		});
+	});
+
+	const generateJwt = async (query: any = {}) => {
+		return encodeSymmetric({
+			qsh: createQueryStringHash({
+				method: "POST",
+				pathname: "/jira/security/workspaces",
+				query
+			}, false),
+			iss: installation.plainClientKey
+		}, await installation.decrypt("encryptedSharedSecret", getLogger("test")));
+	};
+
+	it("Should return a 400 status if no IDs are passed in the body", async () => {
+		app = express();
```

Comment:
existing sampel *.test.ts may not be good, I think just use following should be good
```
const app = getFrontendApp();
await supertest(app).post()...
`

Reply:
And remove 
```
app.use((req, _, next) => {
			req.log = getLogger("test");
			next();
		});
```

Reply:
Ill give that a try! thanks